qiskit_ibm_runtime/base_runtime_job.py

```diff
@@ -0,0 +1,463 @@
+            self._tags = api_response.pop("tags", [])
+            return self._tags
+        else:
+            raise IBMApiError(
+                "An unexpected error occurred when updating the "
+                "tags for job {}. The tags were not updated for "
+                "the job.".format(self.job_id())
+            )
+
+    def properties(self, refresh: bool = False) -> Optional[BackendProperties]:
+        """Return the backend properties for this job.
+
+        Args:
+            refresh: If ``True``, re-query the server for the backend properties.
+                Otherwise, return a cached version.
+
+        Returns:
+            The backend properties used for this job, at the time the job was run,
+            or ``None`` if properties are not available.
+        """
+
+        return self._backend.properties(refresh, self.creation_date)
+
+    def error_message(self) -> Optional[str]:
+        """Returns the reason if the job failed.
+
+        Returns:
+            Error message string or ``None``.
+        """
+        self._set_status_and_error_message()
+        return self._error_message
+
+    def _set_status_and_error_message(self) -> None:
+        """Fetch and set status and error message."""
+        if not self.in_final_state():
+            response = self._api_client.job_get(job_id=self.job_id())
+            self._set_status(response)
+            self._set_error_message(response)
+
+    def _set_status(self, job_response: Dict) -> None:
+        """Set status.
+
+        Args:
+            job_response: Job response from runtime API.
+
+        Raises:
+            IBMError: If an unknown status is returned from the server.
+        """
+        try:
+            reason = job_response["state"].get("reason")
+            reason_code = job_response["state"].get("reason_code")
+            if reason:
+                # TODO remove this in https://github.com/Qiskit/qiskit-ibm-runtime/issues/989
+                if reason.upper() == "RAN TOO LONG":
+                    self._reason = reason.upper()
+                else:
+                    self._reason = reason
+                if reason_code:
+                    self._reason = f"Error code {reason_code}; {self._reason}"
+            self._status = self._status_from_job_response(job_response)
+        except KeyError:
+            raise IBMError(f"Unknown status: {job_response['state']['status']}")
+
+    def _set_error_message(self, job_response: Dict) -> None:
+        """Set error message if the job failed.
+
+        Args:
+            job_response: Job response from runtime API.
+        """
+        if self.errored():
+            self._error_message = self._error_msg_from_job_response(job_response)
+        else:
+            self._error_message = None
+
+    @abstractmethod
+    def _status_from_job_response(self, response: Dict) -> str:
+        """Returns the job status from an API response."""
+        return response["state"]["status"].upper()
```

Comment:
Wasn't sure what the best way to handle this `_status_from_job_response` method was - I don't think it's best practice to make a private method abstract but it makes sense for the child classes to have to implement this method since the `API_TO_JOB_STATUS` is different